R/project_as_ppi.R

```diff
@@ -190,14 +190,19 @@ sample_polar <- function(param, grid_size, range_max, project, ylim, xlim, k = 4
   # get scan parameter indices, and extract data
   index <- polar_to_index(
     cartesian_to_polar(coordinates(gridTopo), elev, k = k, lat = attributes(param)$geo$lat, re = re, rp = rp),
-    attributes(param)$geo$rscale,
-    attributes(param)$geo$ascale
+    rangebin = attributes(param)$geo$rscale,
+    azimbin = attributes(param)$geo$ascale,
+    azimstart = ifelse(is.null(attributes(param)$geo$astart), 0, attributes(param)$geo$astart),
+    rangestart = ifelse(is.null(attributes(param)$geo$rstart), 0, attributes(param)$geo$rstart)
   )
   # set indices outside the scan's matrix to NA
   nrang <- dim(param)[1]
   nazim <- dim(param)[2]
   index$row[index$row > nrang] <- NA
   index$col[index$col > nazim] <- NA
+  # rstart can result in locations outside of the radar scope close to the radar
+  index$row[index$row < 1] <- NA
+  stopifnot(all(index$col >= 1))
```

Comment:
could you add an informative error message that explains why this stop happens and what to do. Is this when rstart has a value that makes certain ranges negative?

Reply:
I think this error basically should not occur it is basically a sanity check. The only reason I could occur I could think of is when some integer float conversions result in just negative numbers (rounding errors). I can also remove it. Or just write that. I think there is no consistent reason why this should occur